### PR TITLE
[Gradle] - Use versions.gradle to aggregate versions across subprojects

### PR DESCRIPTION
**Pull request by jaredsburrows**
Similar to this PR: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/900
@weijjia @iambmelt

### CODE REVIEW

File: msal/src/androidTest/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <string name="app_name">com.microsoft.identity.msal.test</string>
```
Comment (jaredsburrows) [inline msal/src/androidTest/res/values/strings.xml]: I could not build because it was missing `app_name`.

File: msal/build.gradle [hunk 2]
```diff
@@ -103,22 +103,18 @@ task sourcesJar(type: Jar) {
...
+    testCompile "org.mockito:mockito-core:$rootProject.ext.mockitoCoreVersion"
 
     // instrumentation test dependencies
-    androidTestCompile 'com.android.support.test:runner:0.5'
-    // Set this dependency to use JUnit 4 rules
-    androidTestCompile 'com.android.support.test:rules:0.5'
-    androidTestCompile 'org.mockito:mockito-core:1.10.19'
-    androidTestCompile 'com.google.dexmaker:dexmaker:1.2'
-    androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.2'
+    androidTestCompile "com.android.support.test.espresso:espresso-core:$rootProject.ext.espressoCoreVersion"
```
Comment (jaredsburrows) [inline msal/build.gradle]: `espresso-core` includes `runner` and `rules`.

File: msal/src/androidTest/res/values/strings.xml (hunk 1 above)
Comment (iambmelt) [inline msal/src/androidTest/res/values/strings.xml]: We should get this cleaned up. I'll open up an issue and get it fixed.

**Comment by iambmelt** [comment]
Thanks for your contribution and opening up this PR @jaredsburrows! -- I''ll take a look at this asap and see a second reviewer gets eyes on it 👍

File: msal/build.gradle (hunk 2 above)
Comment (weijjia) [inline msal/build.gradle]: We should have explicit dependencies. The sdk is not using espresso for any of the tests, why are we introducing espresso-core?

File: msal/src/androidTest/res/values/strings.xml (hunk 1 above)
Comment (weijjia) [inline msal/src/androidTest/res/values/strings.xml]: Sorry i'm not following, are you saying the tests run are failing or? We didn't see any failures before in the dev branch.

**Comment by jaredsburrows** [comment]
@weijjia It is for android tests only. It is not being introduced to your SDK. `espresso-core` brings in both `runner` and `rules`.

I also could not compile this locally. It was complaining it was missing `app_name`. ADAL has an `app_name` but MSAL did not.

**Comment by weijjia** [comment]
espresso is used for ui testing, i don't think we want to introduce it in the sdk. If you have a reason that why we need to introduce it, happy to know. 

As to the compile, I haven't seen any failures with travis or my local build, Can you attach us the error trace?

**Comment by jaredsburrows** [comment]
@weijjia As I said in my previous comment. It is not introduced in the SDK at all. Please look at the commit, it is in the `androidTestCompile` configuration, not `compile`.

For the `app_name`, it is only the test app. Please look at the previous build here:https://travis-ci.org/AzureAD/microsoft-authentication-library-for-android/builds/256397174.

**Comment by weijjia** [comment]
I'm sorry, test is also part of the sdk. you're introducing espresso as the test dependency, and i don't think we are planning on introducing it. 

For the app_name, if you check the travis history, there was no failures before.

**Comment by jaredsburrows** [comment]
@weijjia I checked out a fresh copy of the `dev` branch and kept `espresso-core` and added back in the `rules` and `runner` dependencies. It seems to build fine locally now.

**Comment by weijjia** [comment]
Thanks for confirming :)

**Comment by jaredsburrows** [comment]
Build passed. Good to merge? :)

**Comment by weijjia** [comment]
Let's wait for Brian to finish the PR, we want two reviewers to approve a PR before merging it.

**Pull request merged**